### PR TITLE
Reset password

### LINKED ISSUE #170: Reset user password
Implement the back-end service for  resetting user's password:
* Send an e-mail with a link to enter a new password
* Reset requests should expire after 24 hours

#### Request a new password
```
POST /users/request_password_reset
{
    "username_or_email": "<username-or-email-address>",
}
```

-> E-Mail with a _Kvm_ link that contains both the e-mail address and a `token` (comma-separated):
```
https://kartevonmorgen.org/#/?reset_password=<email>,<token>
```

#### Reset the existing password

```
POST /users/reset_password
{
    "username_or_email": "<username-or-email-address>",
    "token": "<token>",
    "new_password":"<new-password>"
}
```

### PR DESCRIPTION
**Pull request by flosse**
rebased PR of #180 and resolve #170

### CODE REVIEW

File: src/core/usecases/tests.rs [hunk 1]
```diff
@@ -140,6 +140,23 @@ impl EmailTokenCredentialsRepository for MockDb {
         debug_assert!(len_before >= len_after);
         Ok(len_before - len_after)
     }
+
+    #[cfg(test)]
+    fn get_email_token_credentials_by_email_or_username(
+        &self,
+        email_or_username: &str,
+    ) -> RepoResult<EmailTokenCredentials> {
+        if let Some(token) = self
```
Comment (uklotzde) [inline src/core/usecases/tests.rs]: Unused implementation, replace with `unimplemented!()`

File: src/infrastructure/notify.rs [hunk 2]
```diff
@@ -97,7 +97,8 @@ pub fn user_registered(user: &User, url: &str) {
 
 pub fn user_password_reset_requested(token: &EmailToken) {
     let url = format!(
-        "https://kartevonmorgen.org/#/?reset_password={}",
+        //TODO: "https://kartevonmorgen.org/#/?reset_password={}",
```
Comment (uklotzde) [inline src/infrastructure/notify.rs]: Delete obsolete comment

File: src/infrastructure/notify.rs [hunk 3]
```diff
@@ -97,7 +97,8 @@ pub fn user_registered(user: &User, url: &str) {
...
+        "https://openfairdb.org/password_reset?token={}",
```
Comment (uklotzde) [inline src/infrastructure/notify.rs]: "password_reset" or "password-reset" or "reset-passwort"?

```
#[get("/reset-password?<token>")]
pub fn get_password_reset(flash: Option<FlashMessage>, token: &RawStr) -> Markup {
    view::password_reset(flash, "/users/actions/reset-password", token)
}
```

Very confusing to follow all dependencies and links without a clear naming convention, I'm totally lost.

Comment (flosse) [reply src/infrastructure/notify.rs]: You're totally right! I'm going to change that :)

**Pull request merged**